core/src/main/java/io/questdb/griffin/engine/functions/regex/ILikeStrFunctionFactory.java

```diff
@@ -24,7 +24,98 @@
 
 package io.questdb.griffin.engine.functions.regex;
 
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.SymbolTableSource;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BooleanFunction;
+import io.questdb.griffin.engine.functions.UnaryFunction;
+import io.questdb.griffin.engine.functions.bind.IndexedParameterLinkFunction;
+import io.questdb.griffin.engine.functions.constants.BooleanConstant;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class ILikeStrFunctionFactory extends AbstractLikeStrFunctionFactory {
```

Comment:
There is still a lot of unnecessary code duplication. Please introduce
```
protected abstract boolean isCaseInsensitive();
```
and use this method in the `AbstractLikeStrFunctionFactory`.

Reply:
alright

Reply:
> There is still a lot of unnecessary code duplication. Please introduce
> 
> ```
> protected abstract boolean isCaseInsensitive();
> ```
> 
> and use this method in the `AbstractLikeStrFunctionFactory`.

wait, should this abstract method be declared in `ILikeStrFunctionFactory` class and used in `AbstractLikeStrFunctionFactory` ? but the former extends the later not the other way, so it won't be possible right?

And why does it have a boolean return type? what exactly does it do?

Reply:
This method should belong to `AbstractLikeStrFunctionFactory` and implemented in the child classes. It basically tells whether the factory (and the functions) perform case-insensitive matching or not. Alternatively, you could add an abstract method that would return flags to be provided to `Pattern#compile()`.

Reply:
Tried them both, but the problem is with the `init` ([init](https://github.com/Ln11211/questdb/blob/610034b239e49b1882c37665ad71936c7c76e8ec/core/src/main/java/io/questdb/griffin/engine/functions/regex/AbstractLikeStrFunctionFactory.java#L128)) method which is in a static class `BindLikeStrFunction` and since the declaration of the methods is **abstract**, there is a conflict - **cannot reference non static variable in static context**. Since we are using an abstract method definition they can't be static and cannot be referenced in static context (in this case the bitmask for the flags) etc

Reply:
I'll illustrate what I mean a bit later today.

Reply:
This patch shows what I mean: https://github.com/puzpuzpuz/questdb/commit/c036393fb9f2a96257705154f407892a4d49945c

Reply:
> Did you test it?

You should try running it. Tests should pass.

Reply:
Any option that works for you. You could add my fork as a remote, fetch it and cherrypick the commit; or create a patch from it and apply to your branch; or something else.

Reply:
Sure, that's your PR. Feel free to do any modifications.

Reply:
All done...